components/script/dom/videotracklist.rs

```diff
@@ -0,0 +1,172 @@
+    }
+
+    // TODO(#22799) Integrate DOM Audio and Video track selection with media player
+    pub fn set_selected(&self, idx: usize, value: bool) {
+        if let Some(track) = self.item(idx) {
+            // Ensure a track's selected attribute will change.
+            if track.selected() != value {
+                if track.selected() {
+                    if Some(idx) != self.selected_index() {
+                        // Unselect the current selected track and set the track
+                        // found at the given index as selected.
+                        self.tracks
+                            .borrow()
+                            .iter()
+                            .find(|track| track.selected())
+                            .map(|track| track.set_selected(false));
+
+                        track.set_selected(true);
+                    } else {
+                        // Deselect the currently selected track.
+                        // A task is queued to fire a change event named change
+                        // when no new track is selected after deselecting a track.
+                        track.set_selected(false);
+
+                        let global = &self.global();
+                        let this = Trusted::new(self);
+                        let (source, canceller) = global
+                            .as_window()
+                            .task_manager()
+                            .media_element_task_source_with_canceller();
+
+                        let _ = source.queue_with_canceller(
+                            task!(media_track_change: move || {
+                                let this = this.root();
+                                this.upcast::<EventTarget>().fire_event(atom!("change"));
+                            }),
+                            &canceller,
+                        );
+                    }
+                } else {
+                    // There is no previously selected track and the track found at the
+                    // given index is set to selected.
+                    track.set_selected(true);
+
+                    // Whenever a previously unselected track
+                    // is selected a task is queued to to fire an event named change.
+                    let global = &self.global();
+                    let this = Trusted::new(self);
+                    let (source, canceller) = global
+                        .as_window()
+                        .task_manager()
+                        .media_element_task_source_with_canceller();
+
+                    let _ = source.queue_with_canceller(
+                        task!(media_track_change: move || {
+                            let this = this.root();
+                            this.upcast::<EventTarget>().fire_event(atom!("change"));
+                        }),
+                        &canceller,
+                    );
+                }
+            }
+        }
```

Comment:
I recommend restructuring this code as follows:
```rust
// get chosen track or return early
// if the chosen track's selection status matches the new status, return early
// get the currently selected track and deselect it
// if we are selecting a track, select the chosen track
// queue a change event
```
Does that make sense?

Reply:
Yes thank you. The specification is somewhat confusing on this.